include/miral/miral/internal_client.h

```diff
@@ -55,6 +55,26 @@ class StartupInternalClient
         std::function<void(struct ::wl_display* display)> client_code,
         std::function<void(std::weak_ptr<mir::scene::Session> const session)> connect_notification);
 
+    /// Construct an internal client to be launched when the Mir server starts.
+    ///
+    /// The \p connect_notification call will be called when a connection to the server has
+    /// been secured. This is called on a worker thread and must not block.
+    ///
+    /// The \p client_code callback will be called when the client has been initialized and is ready
+    /// to start interacting with the server. Note that this callback happens on another thread.
+    /// This callback must exit.
+    ///
+    /// The \p join_callback will be called before a join is requested on the client thread so that
+    /// the client may perform some cleanup (e.g. setting a flag to tell the Wayland event loop to stop).
```

Comment:
`join_callback` isn't in the language of the "internal client" code. From that perspective it is `handle_stop` (or similar).